src/Plugins/RestServer/Controllers/v1/TokensController.cs

```diff
@@ -0,0 +1,306 @@
+                .Take(take);
+            if (vaildContracts.Any() == false)
+                return NoContent();
+            var listResults = new List<NEP11TokenModel>();
+            foreach (var contract in vaildContracts)
+            {
+                try
+                {
+                    var token = new NEP11Token(_neosystem, contract.Hash);
+                    listResults.Add(token.ToModel());
+                }
+                catch
+                {
+                }
+            }
+            if (listResults.Any() == false)
+                return NoContent();
+            return Ok(listResults);
+        }
+
+        /// <summary>
+        /// The count of how many Nep-11 contracts are on the blockchain.
+        /// </summary>
+        /// <returns>Count Object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("nep-11/count", Name = "GetNep11TokenCount")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(CountModel))]
+        public IActionResult GetNEP11Count()
+        {
+            return Ok(new CountModel() { Count = NativeContract.ContractManagement.ListContracts(_neosystem.StoreView).Count(ContractHelper.IsNep11Supported) });
+        }
+
+        /// <summary>
+        /// Gets the balance of the Nep-11 contract by an address.
+        /// </summary>
+        /// <param name="sAddressHash" example="0xed7cc6f5f2dd842d384f254bc0c2d58fb69a4761">Nep-11 ScriptHash</param>
+        /// <param name="addressHash" example="0xed7cc6f5f2dd842d384f254bc0c2d58fb69a4761">Neo Address ScriptHash</param>
+        /// <returns>Token Balance Object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("nep-11/{scripthash:required}/balanceof/{address:required}", Name = "GetNep11TokenBalanceOf")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(TokenBalanceModel))]
+        public IActionResult GetNEP11(
+            [FromRoute(Name = "scripthash")]
+            UInt160 sAddressHash,
+            [FromRoute(Name = "address")]
+            UInt160 addressHash)
+        {
+            var contract = NativeContract.ContractManagement.GetContract(_neosystem.StoreView, sAddressHash) ??
+                throw new ContractNotFoundException(sAddressHash);
+            if (ContractHelper.IsNep11Supported(contract) == false)
+                throw new Nep11NotSupportedException(sAddressHash);
+            try
+            {
+                var token = new NEP11Token(_neosystem, sAddressHash);
+                return Ok(new TokenBalanceModel()
+                {
+                    Name = token.Name,
+                    ScriptHash = token.ScriptHash,
+                    Symbol = token.Symbol,
+                    Decimals = token.Decimals,
```

Comment:
Only check `IsNep11Supported(()` and `IsNep17Supported()` is not enough. These methods only check supportedstanard in manifest. So that means, people can create any type of data in decimal() or even make decimal with 2 args. The same to `totalSupply`. If we want list all qulified tokens here, we need check all these methods following https://github.com/neo-project/proposals/blob/master/nep-17.mediawiki and https://github.com/neo-project/proposals/blob/master/nep-11.mediawiki. And catch these invalid type is necessary.
![81c7d66d26f00b85b6ce037ebe3f04d](https://github.com/user-attachments/assets/974af1b9-adab-4c9a-b7c1-238a54bab227)
![f587ccde8776611bc2d63c4f0dcfb8f](https://github.com/user-attachments/assets/9ae8db49-ccf6-4d87-b9c5-7c2d4e929e7c)
![ea61c2d359881b8032404e80958c7e6](https://github.com/user-attachments/assets/a725c7d4-17c1-44df-8dd0-bf90da41ddf9)